src/autocomplete/databases/yql/helpers.ts

```diff
@@ -20,6 +21,36 @@ export const tokenDictionary: TokenDictionary = {
 type AnyRuleInList = (rules: number | number[]) => boolean;
 type AllRulesInList = (rules: number[]) => boolean;
 
+function isFirstPreviousTokenOfType(
+    tokenStream: TokenStream,
+    dictionary: TokenDictionary,
+    tokenIndex: number,
+    tokenType: number,
+): boolean | undefined {
```

Comment:
why do we want to return `undefined`?
let's just return boolean and instead of `undefined` return `false`